subgraphs/compound-v2/src/helpers/account.ts

```diff
@@ -24,3 +30,81 @@ export function AddToLiquidationCount(accountId: string, isLiquidated: Boolean):
   }
   account.save()
 }
+
+export function getOrCreateAccountInProtocol(protocolId: string, accountId: string): AccountInProtocol {
+  const acpId = protocolId.concat('-').concat(accountId);
+  let acp = AccountInProtocol.load(acpId)
+  if (!acp) {
+    acp = new AccountInProtocol(acpId)
+    acp.protocol = protocolId
+    acp.account = accountId
+    acp.depositCount = zeroInt
+    acp.withdrawCount = zeroInt
+    acp.borrowCount = zeroInt
+    acp.repayCount = zeroInt
+    acp.liquidatedCount = zeroInt
+    acp.save()
+  }
+  return acp
+}
+
+export function getOrCreateAccountInMarket(marketId: string, accountId: string): AccountInMarket {
+  const acmId = marketId.concat('-').concat(accountId);
+  let acm = AccountInMarket.load(acmId)
+  if (!acm) {
+    acm = new AccountInMarket(acmId)
+    acm.market = marketId
+    acm.account = accountId
+    acm.deposited = zeroBD
+    acm.borrowed = zeroBD
+    acm.lifetimeBorrowed = zeroBD
+    acm.lifetimeDeposited = zeroBD
+    acm.lifetimeLiquidated = zeroBD
+    acm.lifetimeRepaid = zeroBD
+    acm.lifetimeWithdrawn = zeroBD
+    acm.save()
+  }
+  return acm
+}
+
+export function updateAccountStats(protocolId: string, marketId: string, accountId: string, amount: BigDecimal, eventType: string): void {
+  let acm = getOrCreateAccountInMarket(marketId, accountId)
+  let acp = getOrCreateAccountInProtocol(protocolId, accountId)
+  if (eventType === "DEPOSIT") {
+    acm.depositCount.plus(new BigInt(1))
+    acm.deposited.plus(amount)
+    acm.lifetimeDeposited.plus(amount)
+
+    acp.depositCount.plus(new BigInt(1))
+
+  } else if (eventType === "WITHDRAW") {
+    acm.withdrawCount.plus(new BigInt(1))
+    acm.deposited.minus(amount)
+    acm.lifetimeWithdrawn.plus(amount)
+
+    acp.withdrawCount.plus(new BigInt(1))
+
+  } else if (eventType === "BORROW") {
+    acm.borrowCount.plus(new BigInt(1))
+    acm.borrowed.plus(amount)
+    acm.lifetimeBorrowed.plus(amount)
+
+    acp.borrowCount.plus(new BigInt(1))
+
+  } else if (eventType === "REPAY") {
+    acm.repayCount.plus(new BigInt(1))
+    acm.borrowed.minus(amount)
+    acm.lifetimeRepaid.plus(amount)
+
+    acp.repayCount.plus(new BigInt(1))
+
+  } else if (eventType === "LIQUIDATION") {
+    acm.liquidatedCount.plus(new BigInt(1))
```

Comment:
I'm fairly certain these will not get triggered. 

For some reason I don't remember, `===` does not work as one would expect in Assembly script. I can confirm it didn't work, unless it might have been updated in the last 6 months and I missed the update.

however, i feel it probably has not. because on the `updateMarketStats()` function, where we update fields like `market.deposited` , i noticed that all of the markets still have `0` for all of those fields. so it appears they are not working.

try changing all the `===` to `==` in these if else statements, and i believe it should work.

also, i am not 100% sure if this is working in `updateMarketStats()`

```
    } else if (["REPAY", "LIQUIDAION"].includes(eventType)) {
        market.borrowed.minus(amount)
    }
```
build in functions like `includes()` dont always work, but i dont remember 100%. 
also, i just saw `"LIQUIDAION"` is spelt wrong so that could be affecting it too